src/transformation/mesh_intersection/mesh_intersection.rs

```diff
@@ -256,380 +355,358 @@ fn extract_connected_components(
+    let mut edges = Vec::new();
+    for point_pair in constraints {
+        let p1_id = insert_into_set(point_pair[0].coords, &mut point_set, epsilon);
+        let p2_id = insert_into_set(point_pair[1].coords, &mut point_set, epsilon);
+
+        edges.push([p1_id, p2_id]);
+    }
+
+    let mut points: Vec<_> = point_set.iter().cloned().collect();
+    points.sort_by(|a, b| a.id.cmp(&b.id));
+
+    let tri_points = [tri.a.coords, tri.b.coords, tri.c.coords];
+    let best_source = angle_closest_to_90(&tri_points);
+    let d1 = tri_points[best_source] - tri_points[(best_source + 1) % 3];
+    let d2 = tri_points[(best_source + 2) % 3] - tri_points[(best_source + 1) % 3];
+    let (e1, e2) = planar_gram_schmidt(d1, d2);
+    let project = |p: &Vector3<f64>| spade::Point2::new(e1.dot(p), e2.dot(p));
+
+    // Project points into 2D and triangulate the resulting set.
+
+    let planar_points: Vec<_> = points
+        .iter()
+        .copied()
+        .map(|point| {
+            let point_proj = project(&point.point);
+            spade::Point2::new(point_proj.x, point_proj.y)
+        })
+        .collect();
+    let cdt_triangulation =
+        ConstrainedDelaunayTriangulation::<spade::Point2<f64>>::bulk_load_cdt_stable(
+            planar_points,
+            edges,
+        )
+        .unwrap();
+    debug_assert!(cdt_triangulation.vertices().len() == points.len());
+
+    let points = points.into_iter().map(|p| Point3::from(p.point)).collect();
+    (cdt_triangulation, points)
 }
 
-struct Triangulation {
-    delaunay: ConstrainedDelaunayTriangulation<spade::Point2<Real>>,
-    basis: [Vector<Real>; 2],
-    vtx_handles: [FixedVertexHandle; 3],
-    ref_pt: Point<Real>,
-    ref_proj: [Point2<Real>; 3],
-    normalization: na::Matrix2<Real>,
+// I heavily recommend that this is left here in case one needs to debug the above code.
+fn _mesh_to_obj(mesh: &TriMesh, path: &PathBuf) {
+    let mut file = std::fs::File::create(path).unwrap();
+
+    ObjData {
+        position: mesh
+            .vertices()
+            .into_iter()
+            .map(|v| [v.x as f32, v.y as f32, v.z as f32])
+            .collect(),
+        objects: vec![Object {
+            groups: vec![Group {
+                polys: mesh
+                    .indices()
+                    .into_iter()
+                    .map(|tri| {
+                        SimplePolygon(vec![
+                            IndexTuple(tri[0] as usize, None, None),
+                            IndexTuple(tri[1] as usize, None, None),
+                            IndexTuple(tri[2] as usize, None, None),
+                        ])
+                    })
+                    .collect(),
+                name: "".to_string(),
+                index: 0,
+                material: None,
+            }],
+            name: "".to_string(),
+        }],
+        ..Default::default()
+    }
+    .write_to_buf(&mut file)
+    .unwrap();
```

Comment:
I am not fond of returning the ObjData, the way the obj crate is structured has many flaws. I used it because it was the best I could find but in a perfect world I would have preferred something that separates the geometry data and meta-data information instead of nesting structures into structures. 

My suggestion is to just return an array of positions and indices for now.